### ISSUE TITLE
Bug in root user detection when running under docker

### ISSUE THREAD

**Issue by msheiny** [freedomofpress/ansible-role-grsecurity-build#17]
There is no default `USER` variable set in many docker environments so checking if the user is root by environment var is going to fail. I ended up trying to work around this by injecting it into docker, but accidentally still ran the role as become (which passed the current check) and then had a catastrophic failure at the end of the play because of a weird ansible become with using `fetch` and `become` :)

So tl;dr - Lets gather facts with `setup` in this role (under the assumingly un-privileged user) and then check against `ansible_user_id` instead of relying on env var `USER`
